### PR TITLE
Get job ids with filters

### PR DESCRIPTION
**Pull request by yuzhu**
### What changes are proposed in this pull request?

Use the existing interface of job master to allow filtering of jobs by name and by job status

### Why are the changes needed?
Currently, there is no way for client to efficiently look up job status and find incomplete jobs with a specific name/type
This leads to O(n) rpc calls from replicationChecker every heartbeat.

### Does this PR introduce any user facing changes?
No

### CODE REVIEW

**Comment by yuzhu** [comment]
@apc999 obvious need to add more unit test and integration tests, but want to get an early round of review as I continue to improve this.

File: job/client/src/main/java/alluxio/client/job/JobMasterClient.java [hunk 1]
```diff
@@ -75,9 +76,10 @@ public static JobMasterClient create(JobMasterClientContext conf) {
   JobServiceSummary getJobServiceSummary() throws IOException;
 
   /**
+   * @param option list options
    * @return the list of ids of all jobs
    */
-  List<Long> list() throws IOException;
```
Comment (apc999) [inline job/client/src/main/java/alluxio/client/job/JobMasterClient.java:80]: shall we have `default  List<Long> list()`?

**Review by apc999 — COMMENTED** [review]
minor comment left.

**Comment by yuzhu** [comment]
@apc999 addressed comment, PTAL.

**Comment by yuzhu** [comment]
@apc999 will address any further comments in a separate PR. merging this for now.

**Comment by yuzhu** [comment]
alluxio-bot, merge this please

**Comment by yuzhu** [comment]
alluxio-bot, cherry-pick this to branch-2.6 please

**Pull request merged**